src/SqlAsyncCollector.cs

```diff
@@ -237,22 +252,57 @@ private async Task UpsertRowsAsync(IEnumerable<T> rows, SqlAttribute attribute,
         /// to upsert to and returns the extra property names in a List.
         /// </summary>
         /// <param name="columns"> The columns of the table to upsert to </param>
+        /// <param name="rows"> The rows to be upserted </param>
         /// <returns>List of property names that don't exist in the table</returns>
-        private static IEnumerable<string> GetExtraProperties(IDictionary<string, string> columns)
+        private static IEnumerable<string> GetExtraProperties(IDictionary<string, string> columns, IEnumerable<T> rows)
         {
+            if (typeof(T) == typeof(JObject))
+            {
+                var jsonObj = JObject.Parse(rows.First().ToString());
```

Comment:
What if there aren't any rows returned?

Reply:
It would come to this if rows doesn't have any data. There is a check in the parent method ->  if (this._rows.Count != 0)